### PR TITLE
[master] Choreo Connect graphQL documentation addition

### PR DESCRIPTION
**Pull request by slahirucd7**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: en/docs/includes/deploy/cc-deploy-api-standalone-mode.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-### Step 3 - Add a Choreo Connect Environment to apictl
+git### Step 3 - Add a Choreo Connect Environment to apictl
```
Comment (renuka-fernando) [inline en/docs/includes/deploy/cc-deploy-api-standalone-mode.md:1]:
```suggestion
### Step 3 - Add a Choreo Connect Environment to apictl
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+- API life-cycle management
+
+!!! info "Before you begin"
+
+    - This guide assumes that you already have a up and running Choreo Connect instance configured with WSO2 API Manager. If not, checkout the [Quick Start Guide]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/quick-start-guide-docker-with-apim/) on how to install and run Choreo Connect. To learn more about Choreo Connect, have a look at the [Overview of Choreo Connect]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/choreo-connect-overview).
+    - Also you need to have a correctly defined GraphQL schema file.
+
+### Step 1 - Create a GraphQL API in API Manager
+
+1. Create a GraphQL API by following the steps in [Create a GraphQL API]({{base_path}}/design/create-api/create-a-graphql-api/#create-a-graphql-api).
```
Comment (slahirucd7) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md:31]: Mention about the two steps.

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+5. Open the **Subscriptions** tab from the left menu bar, click on **PROD KEYS**, and generate keys.
+
+6. Open the **Try Out** tab and click **GET TEST KEY**. This will include the access token in the cURL command you generate in the section below.
+
+!!! tip 
+    To generate a temporary test key to invoke the API, follow the steps [here]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/security/generate-a-test-jwt/)
+
+### Step 4 - Invoke the API by providing a valid GraphQL query
+
+!!! note
```
Comment (slahirucd7) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md:57]: Remove note section.

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+- Import and download GraphQL schema file
+- Runtime configurations
+    - Transport level security
+    - [GraphQL query analysis]({{base_path}}/design/rate-limiting/graphql-api/overview-query-limits-for-graphql/#static-query-analyzer)
+- [Rate limiting policies (API level and operation level)]({{base_path}}/design/create-api/create-a-graphql-api/#rate-limiting-for-graphql-operations)
+- [Security scopes]({{base_path}}/design/create-api/create-a-graphql-api/#authorization-for-graphql-operations)
+- Enable/ disable security for GraphQL operations
+- Deploy option for multiple gateways
+- API life-cycle management
+
```
Comment (slahirucd7) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md:23]: Move supported operations Note to here.

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+The CLI tool ([**apictl**]({{base_path}}/install-and-setup/setup/api-controller/getting-started-with-wso2-api-controller/#download-and-initialize-the-apictl)) does not support initializing projects for Streaming APIs yet. However, you can download a GraphQL API from a WSO2 API Manager instance and deploy the downloaded project in Choreo Connect. You do not need any special configurations in API-M or Choreo Connect for this. If you do not have a running API Manager instance already, simply download the pack from the [official site](https://wso2.com/api-manager/) and start one locally.
+
+The following are the steps to download and deploy the API using apictl.
+
+### Step 1 - Create an API 
+
+Create a GraphQL API following the steps in [Create a GraphQL API in API Manager](#step-1-create-a-graphql-api-in-api-manager). 
+
+### Step 2 - Download the API
```
Comment (slahirucd7) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md:97]: Use a project uploaded to the CC public repo

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+After creating the API and making the necessary changes, click **Download API** in the top right corner. The API will be downloaded as a .zip file and after that it can be deployed via apictl using the next few commands.
+
+{! ./includes/deploy/cc-deploy-api-standalone-mode.md !}
```
Comment (slahirucd7) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md:101]: Explain the requirement to add CC with APICTL.
https://github.com/wso2/docs-apim/pull/6072/files#diff-3824711a0495de826c9c78e6712b7f58bc5c7a4a7d6148e21b8387d8c8d06dd8R10

Comment (slahirucd7) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md:101]: remove include.

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+## Via WSO2 API Manager Publisher Portal
+
+Below instructions explain how to deploy a GraphQL to the Choreo Connect via the Publisher Portal in WSO2 API Manager. The Publisher Portal provides following capabilities for GraphQL APIs.
+
+- Import and download GraphQL schema file
+- Runtime configurations
+    - Transport level security
+    - [GraphQL query analysis]({{base_path}}/design/rate-limiting/graphql-api/overview-query-limits-for-graphql/#static-query-analyzer)
+- [Rate limiting policies (API level and operation level)]({{base_path}}/design/create-api/create-a-graphql-api/#rate-limiting-for-graphql-operations)
```
Comment (slahirucd7) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md:18]: With support of WSO2 API-M.

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+|**Mode**                               | **Method**    |
+|---------------------------------------|---------------|
+|[Choreo Connect with WSO2 API Manager as a Control Plane]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/concepts/apim-as-control-plane/)   | [Via WSO2 API Manager Publisher Portal](#via-wso2-api-manager-publisher-portal)  |
+|[Choreo Connect as a Standalone Gateway]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/concepts/as-a-standalone-gateway/)                  |[Via apictl for Standalone Mode](#via-apictl-for-standalone-mode)                |
+
+## Via WSO2 API Manager Publisher Portal
+
+Below instructions explain how to deploy a GraphQL to the Choreo Connect via the Publisher Portal in WSO2 API Manager. The Publisher Portal provides following capabilities for GraphQL APIs.
+
+- Import and download GraphQL schema file
```
Comment (slahirucd7) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md:14]: Move this to top section.

Comment (slahirucd7) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md:14]: Remove mentioning about publisher portal.

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md [hunk 9]
```diff
@@ -0,0 +1,293 @@
...
+|[Choreo Connect as a Standalone Gateway]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/concepts/as-a-standalone-gateway/)                  |[Via apictl for Standalone Mode](#via-apictl-for-standalone-mode)                |
+
+## Via WSO2 API Manager Publisher Portal
+
+Below instructions explain how to deploy a GraphQL to the Choreo Connect via the Publisher Portal in WSO2 API Manager.
+
+!!! info "Before you begin"
+
+    - This guide assumes that you already have a up and running Choreo Connect instance configured with WSO2 API Manager. If not, you can refer to the [Quick Start Guide]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/quick-start-guide-docker-with-apim/) on how to run Choreo Connect with WSO2 API Manager. To learn more about Choreo Connect related componentes, refer to the [Overview of Choreo Connect]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/choreo-connect-overview).
+    - GraphQL schema file (SDL) and a GraphQL server implementaion relevant to the SDL.
```
Comment (suksw) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md:37]: Found https://apim.docs.wso2.com/en/latest/assets/attachments/learn/schema_graphql.graphql in https://apim.docs.wso2.com/en/latest/tutorials/create-and-publish-a-graphql-api/ . Shall we point it here as well? and we can also bring the info about the sample backend in line 126 here.

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md [hunk 10]
```diff
@@ -0,0 +1,293 @@
...
+          <td>
+             <p>1.0.0</p>
+          </td>
+          </tr>
+          <tr>
+          <td>
+             <p>Endpoint</p>
+          </td>
+          <td>
+             <a href="http://localhost:8080/graphql" target="_blank">http://localhost:8080/graphql</a>
```
Comment (suksw) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md:117]: Shall give something like http://backend:8080/graphql and mention the etc mapping here?

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md [hunk 11]
```diff
@@ -0,0 +1,293 @@
...
+    curl -X POST "<Docker-hostname>:<Docker-port>/<API-context>/<API-version>" -H "Authorization: Bearer $TOKEN" -d "<query>" -H "Content-Type: application/json" -k 
+    ```
+
+    ```bash tab="Example"
+    curl -X POST "https://localhost:9095/starwars/1.0.0" -H "Authorization: Bearer $TOKEN" -d '{"query":"query MyQuery { allDroids { appearsIn } }","variables":null}' -H "Content-Type: application/json" -k 
+    ```
+
+    ```bash tab="Response"
+    {"data":{"allDroids":[{"appearsIn":["NEWHOPE","EMPIRE","JEDI"]},{"appearsIn":["NEWHOPE","EMPIRE","JEDI"]}]}}
+    ```
```
Comment (suksw) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md:293]: Shall we add https://apim.docs.wso2.com/en/latest/design/rate-limiting/graphql-api/overview-query-limits-for-graphql/ in a "see also" section?

Comment (slahirucd7) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md:293]: already linked this in top section

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md [hunk 12]
```diff
@@ -114,7 +114,7 @@ You can create a GraphQL API in WSO2 API-M publisher portal by following the ste
              <p>Endpoint</p>
           </td>
           <td>
-             <a href="http://localhost:8080/graphql" target="_blank">http://localhost:8080/graphql</a>
+             <a href="http://host.docker.internal:8080/graphql" target="_blank">http://host.docker.internal:8080/graphql</a>
```
Comment (suksw) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md:117]: This won't work for linux right?

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md [hunk 13]
```diff
@@ -182,24 +177,23 @@ You can create a GraphQL API in WSO2 API-M publisher portal by following the ste
...
 in `api.yaml`, `graphql-complexity.yaml` and `definitions/schema.graphql` files. There is an explanation regarding those attribute values in 
 [here](https://github.com/wso2/product-microgateway/tree/main/samples/apiProjects/apiProjects/SampleGraphQLApi/README.md).
 
-The following steps describe how to deploy GraphQL API in Choreo Connect standalone mode.
+The following steps describe how to deploy a GraphQL API in Choreo Connect standalone mode.
 
 ### Step 1 - Create an API Project
 
-When using GraphQL APIs in Choreo Connect standalone mode, you need to have an API project. Structure of a sample GraphQL API project included in below.
+When using GraphQL APIs in Choreo Connect standalone mode, you need to have an API project. The tructure of a sample GraphQL API project is included in below.
```
Comment (suksw) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md:196]: typo `structure`

**Review by pubudu538 — APPROVED** [review]

**Review by samgnaniah — APPROVED** [review]

**Pull request merged**